src/hooks/room/useRoomThreadNotifications.ts

```diff
@@ -0,0 +1,66 @@
+/*
+Copyright 2023 The Matrix.org Foundation C.I.C.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import { NotificationCountType, Room, RoomEvent, ThreadEvent } from "matrix-js-sdk/src/matrix";
+import { useCallback, useEffect, useState } from "react";
+
+import { NotificationColor } from "../../stores/notifications/NotificationColor";
+import { doesRoomOrThreadHaveUnreadMessages } from "../../Unread";
+import { useEventEmitter } from "../useEventEmitter";
+
+/**
+ * Tracks the thread unread state for an entire room
+ * @param room the room to track
+ * @returns the type of notification for this room
+ */
+export const useRoomThreadNotifications = (room: Room): NotificationColor => {
```

Comment:
I wonder if logic like this (and the call status…) should not better go to a store or something else, that emits events. A hook could then just couple it to the component. If we start using these hooks in many places we will add quite a number of the same event listeners and code execution. This can become heavy, esp. in loops.

Probably not for this PR, but rather for general discussion.